vision_agent_tools/tools/text_to_image.py

```diff
@@ -0,0 +1,76 @@
+from enum import Enum
+
+from PIL import Image
+
+from vision_agent_tools.models.flux1 import Flux1Task, Flux1Config
+from vision_agent_tools.models.model_registry import get_model_class
+from vision_agent_tools.shared_types import BaseTool, Device
+
+
+class TextToImageModel(str, Enum):
+    FLUX1 = "flux1"
+
+
+class TextToImage(BaseTool):
+    """
+    Tool to perform image generation or image mask inpainting using text prompts and the specified ML model
+    """
+
+    def __init__(
+        self,
+        model: TextToImageModel,
+        model_config: Flux1Config | None = None,
+    ):
+        if model not in TextToImageModel._value2member_map_:
+            raise ValueError(
+                f"Model '{model}' is not a valid model for {self.__class__.__name__}."
+            )
+
+        self._model_name: TextToImageModel = model
+
+        if model == TextToImageModel.FLUX1:
+            self._model_config = model_config or Flux1Config()
+            self._model_class = get_model_class(model_name=self._model_name)
+            model_instance = self._model_class()
+            super().__init__(model=model_instance(self._model_config))
+        else:
+            raise ValueError(f"Model is not supported: '{model}'")
+
+    def __call__(
+        self,
+        task: Flux1Task,
```

Comment:
same here, I think to avoid passing an unsupported task we could use `@validate_arguments`

Reply:
Resolved!